querier.cc

```diff
@@ -339,41 +356,67 @@ std::optional<Querier> querier_cache::lookup_querier(
     tracing::trace(trace_state, "Dropping querier because {}", cannot_use_reason(can_be_used));
     ++stats.drops;
 
+    // Save semaphore name and address for later to use it in
+    // error/warning message
+    auto q_semaphore_name = q.permit().semaphore().name();
+    auto q_semaphore_address = reinterpret_cast<uintptr_t>(&q.permit().semaphore());
+
     // Close and drop the querier in the background.
     // It is safe to do so, since _closing_gate is closed and
     // waited on in querier_cache::stop()
     (void)with_gate(_closing_gate, [q = std::move(q)] () mutable {
         return q.close().finally([q = std::move(q)] {});
     });
 
+    if (can_be_used == can_use::no_scheduling_group_mismatch) {
+        ++stats.scheduling_group_mismatches;
```

Comment:
@denesb should I also decrease `stats.drops`. Because now it will be increased if semaphore mismatch happens, but [drops' description](https://github.com/scylladb/scylladb/blob/master/replica/database.cc#L576) says: ```Counts querier cache lookups that found a cached querier but had to drop it due to position mismatch```

Reply:
I think we should extend the comment instead. Drop should include all reasons for dropping a querier.